demo/vagrant/jobs/webapp.nomad

```diff
@@ -50,14 +46,82 @@ job "webapp" {
       }
 
       service {
-        name = "webapp"
+        name = "demo-webapp"
         port = "http"
 
         check {
           type     = "http"
           path     = "/"
-          interval = "2s"
-          timeout  = "2s"
+          interval = "3s"
+          timeout  = "1s"
+        }
+      }
+    }
+
+    task "toxiproxy" {
+      driver = "docker"
+
+      lifecycle {
+        hook    = "prestart"
+        sidecar = true
+      }
+
+      config {
+        image      = "shopify/toxiproxy:2.1.4"
+        entrypoint = ["/entrypoint.sh"]
+
+        volumes = [
+          "local/entrypoint.sh:/entrypoint.sh",
+        ]
+
+        port_map = {
+          api = 8474
+        }
+      }
+
+      template {
+        data = <<EOH
+#!/bin/sh
+
+set -ex
+
+/go/bin/toxiproxy -host 0.0.0.0  &
+/go/bin/toxiproxy-cli create webapp -l 0.0.0.0:{{ env "NOMAD_PORT_webapp"  }} -u {{ env "NOMAD_ADDR_webapp_http" }}
+/go/bin/toxiproxy-cli toxic add -n latency -t latency -a latency=1000 -a jitter=500 webapp
+tail -f /dev/null
```

Comment:
This is awful but I couldn't find a way to bring back the `toxiproxy` process (`fg` didn't work). Any ideas?

Reply:
The inability to configure the toxics via a config file makes me sad.
I can't think of a better way currently; i'll keep thinking about it though.

Reply:
one alternative is an init task that performs the toxiproxy-cli commands. it looks like this: 
```
    task "toxiproxy-config" {
      driver = "docker"

      lifecycle {
        hook    = "prestart"
        sidecar = false
      }

      config {
        image      = "shopify/toxiproxy:2.1.4"
        entrypoint = ["/entrypoint.sh"]

        volumes = [
          "local/entrypoint.sh:/entrypoint.sh",
        ]
      }

      template {
        data = <<EOH
#!/bin/sh

set -ex

while ! nc -z $NOMAD_IP_toxiproxy_api $NOMAD_PORT_toxiproxy_api; do
  echo "Waiting toxiproxy to launch on $NOMAD_PORT_toxiproxy_api..."
  sleep 2
done

echo "toxiproxy is listening"

/go/bin/toxiproxy-cli -h $NOMAD_ADDR_toxiproxy_api create webapp -l 0.0.0.0:{{ env "NOMAD_PORT_webapp"  }} -u {{ env "NOMAD_ADDR_webapp_http" }}
/go/bin/toxiproxy-cli -h $NOMAD_ADDR_toxiproxy_api toxic add -n latency -t latency -a latency=1000 -a jitter=500 webapp
        EOH

        destination = "local/entrypoint.sh"
        perms       = "755"
      }

      resources {
        cpu    = 100
        memory = 32
      }
    }
```

then toxiproxy sidecar can run like usual.

Reply:
in my testing this task sometimes fails, because toxiproxy will apparently start listening before it's ready to run, but then the init is restarted. the nomad restart interval for this task may need some tuning.

Reply:
Hum...I might need to sprinkle some magic `sleep` here and there. I will try to reproduce it.

Reply:
I've used `wget` instead of `nc` to make sure `toxiproxy` is ready. 

I also found a potential bug in Nomad where allocs in a deployment could be marked as `unhealthy` even though they passed the checks. This could be problem you were experiencing.